modules/game_market/classes/t_history.lua

```diff
@@ -0,0 +1,286 @@
+	  local color = ((index % 2 == 0) and '#414141' or '#484848')
+	  widget:setId(color)
+	  widget.actionId = index
+	  widget:setBackgroundColor(color)
+	  widget:setColor('#c0c0c0')
+	  widget.amount:setText(data.amount)
+	  widget.name:setText(g_things.getThingType(data.itemId, ThingCategoryItem):getMarketData().name)
+	  widget.endAt:setText(os.date("%Y-%m-%d, %H:%M:%S", data.timestamp))
+	  widget.status:setText(MarketBuyStatus[data.state])
+	  widget.piecePrice:setColor("#c0c0c0")
+	  widget.totalPrice:setColor("#c0c0c0")
+	  widget.name:setColor("#c0c0c0")
+	  widget.amount:setColor("#c0c0c0")
+	  widget.endAt:setColor("#c0c0c0")
+	  widget.status:setColor("#c0c0c0")
+
+	  if data.itemTier > 0 then
+		  widget.name:setText(widget.name:getText() .. " (Tier " .. data.itemTier .. ")")
+	  end
+
+	  local totalPrice = data.price * data.amount
+	  local unitPrice = data.price
+	  widget.piecePrice:setText(convertLongGold(unitPrice))
+	  widget.totalPrice:setText(convertLongGold(totalPrice))
+	  if totalPrice > 99999999 then
+		  widget.totalPrice:setTooltip(comma_value(totalPrice))
+	  end
+
+	  if unitPrice > 99999999 then
+		  widget.piecePrice:setTooltip(comma_value(unitPrice))
+	  end
+	end
+
+	local window = marketWindow.MarketHistory:recursiveGetChildById('buyOffersList')
+	window:focusChild(nil)
+	lastSelectedHistoryBuy = nil
+end
+
+function MarketHistory.onParseMarketHistory(buyOffers, sellOffers)
+	local window = marketWindow.MarketHistory.offerHistory
+	window.sellOffersList:destroyChildren()
+	window.buyOffersList:destroyChildren()
+	lastSelectedHistorySell = nil
+	lastSelectedHistoryBuy = nil
+
+	topListFitItems = math.floor(window.sellOffersList:getHeight() / labelSize)
+	topListMin = 0
+	topListPool = {}
+	topListData = sellOffers
+	topListMax = #sellOffers
+
+	for i = 1, historyOfferPool do
+		local widget = g_ui.createWidget('MarketHistoryWidget', window.sellOffersList)
+		local data = sellOffers[i]
+		if not data then
+			widget:setVisible(false)
+			goto continue
+		end
+
+		local color = i % 2 == 0 and '#414141' or '#484848'
+		widget:setId(color)
+		widget.actionId = i
+		widget:setBackgroundColor(color)
+		widget.amount:setText(data.amount)
+		widget.name:setText(g_things.getThingType(data.itemId, ThingCategoryItem):getMarketData().name)
+		widget.endAt:setText(os.date("%Y-%m-%d, %H:%M:%S", data.timestamp))
+		widget.status:setText(MarketSellStatus[data.state])
+
+		if data.itemTier > 0 then
+			widget.name:setText(widget.name:getText() .. " (Tier " .. data.itemTier .. ")")
+		end
+
+		local totalPrice = data.price * data.amount
+		local unitPrice = data.price
+		widget.piecePrice:setText(convertLongGold(unitPrice))
+		widget.totalPrice:setText(convertLongGold(totalPrice))
+		if totalPrice > 99999999 then
+		  widget.totalPrice:setTooltip(comma_value(totalPrice))
+		end
+
+		if unitPrice > 99999999 then
+		  widget.piecePrice:setTooltip(comma_value(unitPrice))
+		end
+
+		table.insert(topListPool, widget)
+		:: continue ::
```

Comment:
```suggestion
		:: continue ::
```
this breaks lua 5.1, never use :: continue ::

Reply:
Nothing is breaking, all working good at this part, can you show me the error?

Reply:
He doesnt say it breaks the code. Its just a standard. No modern langusges should use continue or goto. There are better options. 

This code smells bad to me

Reply:
Got it, will fix that.